### DISCUSSION TITLE
Improve the VideoPlayer [Looping, Playback Speed,  MP4 Support]

### DISCUSSION THREAD

**Discussion by ReddKarma** [godotengine/godot-proposals#5909] [GUI]
MP4 / MOV: Currently it's recommended to use OGV files when importing videos which isn't a very popular video format doesn't provide much video quality. So adding support for more popular formats like MP4 or MOV would make development easier.

Video Looping: It suprised me when I realized there was no boolean option to loop a video after it's done playing. Currently the method involves waiting for a signal that fires once the video has ended and then replaying the video which though easy to do I find it to be a "hacky" work around.

Playback Speed: Being able to manipulate the speed in which videos are played feels like a must have feature.

That about wraps it. I'm excited for the official release of Godot 4 and thank you guys for your hardwork.

**Comment by Calinou** [comment]
Support for more video formats won't be tackled in core, but will have to be implemented with extensions instead. See https://github.com/godotengine/godot-proposals/issues/3286 for rationale.

Support for looping is already being tracked in https://github.com/godotengine/godot-proposals/issues/264.
